### ISSUE TITLE
possible to have the size limit set by memory size instead of memory items?

### ISSUE THREAD

**Issue by ouvaa** [elastic/go-freelru#23]
1. possible to have the size limit set by memory size instead of memory items?
2. possible to fully utilize the memory storage efficiency by it's size? e.g. 1byte takes 1byte + meta info
10 bytes takes 10bytes + meta info etc.

**Comment by rockdaboot** [comment]
> possible to have the size limit set by memory size instead of memory items?

That is possible. The number of items is `memsize / (sizeof(key) + sizeof(value) + 32)`.
So for now, you can use this calculation. I'd possibly wait for some upvotes before adding a new API call.

> possible to fully utilize the memory storage efficiency by it's size? e.g. 1byte takes 1byte + meta info 10 bytes takes 10bytes + meta info etc

I am not sure that I understand correctly. What exactly is "meta info"?

**Comment by ouvaa** [comment]
@rockdaboot 
1. i understand the calculation but my key size and value is variably defined. it's not consistent 1 length.
2. meta info is the 32 byte overhead of each cache as you have mentioned.

where to do the upvotes to add the api call? thx in advance.

**Comment by rockdaboot** [comment]
> i understand the calculation but my key size and value is variably defined. it's not consistent 1 length.

This sounds you are using `any` or byte slices or other pointer types as type for keys and values. And if I understand correctly, you want FreeLRU to track the memory pointed to? It sounds like this requires a bigger change in the implementation. Likely there are implementations that do exactly this!?

It would be great if you could come up with a PoC as a base for further discussion (I am still nor sure if we are on the same page).

**Comment by ouvaa** [comment]
@rockdaboot yes i'm using []byte a lot and you only have "string" because that's the only way to do "comparison"?

anyway, https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/lib/workingsetcache/cache.go

it's "lru" too
